### ISSUE TITLE
Unknown packet XX from server 

### ISSUE THREAD

**Issue by vlstv** [ClickHouse/ClickHouse#21588]
Some queries to distributed table fail with  this error
`
2021.03.10 09:55:07.000828 [ 8466 ] {88a44f07-ba05-443a-88cc-14fd260b8297} <Error> executeQuery: Code: 100, e.displayText() = DB::Exception: Unknown packet 97 from server node2-dal13-sl.systems:9000 (version 21.2.5.5 (official build)) (from 54.208.201.221:35979) (in query: SELECT domain AS domain, url AS url, uri AS uri, ip AS ip, method AS method, status_code AS status_code, conviction__rule_name AS rule_name, country AS country, computed_action AS action, conviction__rule_id AS rule_id, id AS id, ua AS ua, ua_parsed__client AS ua_parsed_client, organization AS organization, toInt64(timestamp) AS timestamp, conviction__reference_id AS reference_id, result AS result, traffic_type AS traffic_type FROM default.logs WHERE (((bitTest(traffic_type_query, 1) = 1)) AND domain='82deae48a65148a881b4bdfbd4c5deec.stackpathdns.com' AND timestamp>=1615283645000 AND timestamp<1615370045000) ORDER BY timestamp DESC LIMIT 100 FORMAT JSONCompact ), Stack trace (when copying this message, always include the lines below):

0. DB::Connection::receivePacket(std::__1::function<void (Poco::Net::Socket&)>) @ 0xf517cc5 in /usr/bin/clickhouse
1. DB::MultiplexedConnections::receivePacketUnlocked(std::__1::function<void (Poco::Net::Socket&)>) @ 0xf525b75 in /usr/bin/clickhouse
2. DB::MultiplexedConnections::drain() @ 0xf5262b0 in /usr/bin/clickhouse
3. DB::RemoteQueryExecutor::finish(std::__1::unique_ptr<DB::RemoteQueryExecutorReadContext, std::__1::default_delete<DB::RemoteQueryExecutorReadContext> >*) @ 0xe6ad867 in /usr/bin/clickhouse
4. DB::PipelineExecutor::tryAddProcessorToStackIfUpdated(DB::ExecutingGraph::Edge&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, unsigned long) @ 0xf63ad1c in /usr/bin/clickhouse
5. DB::PipelineExecutor::prepareProcessor(unsigned long, unsigned long, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node
*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::unique_lock<std::__1::mutex>) @ 0xf63b4e9 in
 /usr/bin/clickhouse
6. DB::PipelineExecutor::executeStepImpl(unsigned long, unsigned long, std::__1::atomic<bool>*) @ 0xf63e9f0 in /usr/bin/clickhouse
7. ? @ 0xf642f26 in /usr/bin/clickhouse
8. ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0x84f700f in /usr/bin/clickhouse
9. ? @ 0x84faaa3 in /usr/bin/clickhouse
10. start_thread @ 0x7ea5 in /usr/lib64/libpthread-2.17.so
11. clone @ 0xfe96d in /usr/lib64/libc-2.17.so
`
you can see query example in the error message
default.logs is a distributed table. We have 3 nodes cluster running clickhouse-server version 21.2.5.5

**Comment by maxistua** [comment]
I have the same problem:

```
Code: 100, e.displayText() = DB::Exception: Unknown packet 0 from server XXX.XXX.XXX.XXX:9000 (version 21.2.5.5 (official build)) (from 127.0.0.1:43620) (in query: SELECT min(__version), max(__version) FROM (select __version FROM lake.storage_history WHERE __version > 1615365398180181313 ORDER BY __version LIMIT 300000)), Stack trace (when copying this message, always include the lines below):

0. DB::Connection::receivePacket(std::__1::function<void (Poco::Net::Socket&)>) @ 0xf517cc5 in /usr/bin/clickhouse
1. DB::MultiplexedConnections::receivePacketUnlocked(std::__1::function<void (Poco::Net::Socket&)>) @ 0xf525b75 in /usr/bin/clickhouse
2. DB::MultiplexedConnections::drain() @ 0xf5262b0 in /usr/bin/clickhouse
3. DB::RemoteQueryExecutor::finish(std::__1::unique_ptr<DB::RemoteQueryExecutorReadContext, std::__1::default_delete<DB::RemoteQueryExecutorReadContext> >*) @ 0xe6ad867 in /usr/bin/clickhouse
4. DB::PipelineExecutor::tryAddProcessorToStackIfUpdated(DB::ExecutingGraph::Edge&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, unsigned long) @ 0xf63ad1c in /usr/bin/clickhouse
5. DB::PipelineExecutor::prepareProcessor(unsigned long, unsigned long, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::unique_lock<std::__1::mutex>) @ 0xf63b4e9 in /usr/bin/clickhouse
6. DB::PipelineExecutor::tryAddProcessorToStackIfUpdated(DB::ExecutingGraph::Edge&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, unsigned long) @ 0xf63adbc in /usr/bin/clickhouse
7. DB::PipelineExecutor::prepareProcessor(unsigned long, unsigned long, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::unique_lock<std::__1::mutex>) @ 0xf63b4e9 in /usr/bin/clickhouse
8. DB::PipelineExecutor::executeStepImpl(unsigned long, unsigned long, std::__1::atomic<bool>*) @ 0xf63e9f0 in /usr/bin/clickhouse
9. ? @ 0xf642f26 in /usr/bin/clickhouse
10. ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0x84f700f in /usr/bin/clickhouse
11. ? @ 0x84faaa3 in /usr/bin/clickhouse
12. start_thread @ 0x7ea5 in /usr/lib64/libpthread-2.17.so
13. __clone @ 0xfe9fd in /usr/lib64/libc-2.17.so
```

Context:
We have 4 nodes with version 21.2.5.5
lake.storage_history is distributed table

Some message from trace log:
```
[new.rs1.sh2.XXXX] 2021.03.11 10:48:39.732270 [ 23133 ] {224d5f9c-6cd5-4316-81cf-d754d56f0125} <Trace> StorageDistributed (storage_history): (XXX.XXX.XXX.XXX:9000) Cancelling query because enough data has been read

[new.rs1.sh2.XXXX] 2021.03.11 10:48:39.819608 [ 23764 ] {224d5f9c-6cd5-4316-81cf-d754d56f0125} <Error> executeQuery: Code: 100, e.displayText() = DB::Exception: Unknown packet 0 from server XXX.XXX.XXX.XXX:9000 (version 21.2.5.5 (official build)) (from 127.0.0.1:43620) (in query: SELECT min(__version), max(__version) FROM (select __version FROM lake.storage_history WHERE __version > 1615365398180181313 ORDER BY __version LIMIT 300000)), Stack trace (when copying this message, always include the lines below):
```

It seems that when the distributed table has already collected enough rows for processing, the server sends a cancellation request to those nodes from which it has not yet received a response. But immediately after that, the server receives response, which leads to this error.

In version 21.2.2.8 of the clickhouse-server, this request generated a fatal error
```
2021.03.11 09:06:57.693464 [ 8085 ] {} <Fatal> BaseDaemon: ########################################
2021.03.11 09:06:57.693680 [ 8085 ] {} <Fatal> BaseDaemon: (version 21.2.2.8 (official build), build id: 67D2DA03ADA750F18CEB331C65F50D0F56E33713) (from thread 5241) (query_id: fbb793ca-80ed-4761-aa69-6fcfac4996
9a) Received signal Segmentation fault (11)
2021.03.11 09:06:57.693819 [ 8085 ] {} <Fatal> BaseDaemon: Address: 0x90 Access: read. Address not mapped to object.
2021.03.11 09:06:57.693929 [ 8085 ] {} <Fatal> BaseDaemon: Stack trace: 0xf52fe04 0xf52fb29 0xf53d6b5 0xf53ddf0 0xe6b4657 0xf6527fc 0xf652fc9 0xf65289c 0xf652fc9 0xf6564d0 0xf65aa06 0x851158f 0x8515023 0x7fa954d
7eea5 0x7fa954aa79fd
2021.03.11 09:06:57.694413 [ 8085 ] {} <Fatal> BaseDaemon: 2. ? @ 0xf52fe04 in /usr/bin/clickhouse
2021.03.11 09:06:57.694554 [ 8085 ] {} <Fatal> BaseDaemon: 3. DB::Connection::receivePacket(std::__1::function<void (Poco::Net::Socket&)>) @ 0xf52fb29 in /usr/bin/clickhouse
2021.03.11 09:06:57.694660 [ 8085 ] {} <Fatal> BaseDaemon: 4. DB::MultiplexedConnections::receivePacketUnlocked(std::__1::function<void (Poco::Net::Socket&)>) @ 0xf53d6b5 in /usr/bin/clickhouse
2021.03.11 09:06:57.694765 [ 8085 ] {} <Fatal> BaseDaemon: 5. DB::MultiplexedConnections::drain() @ 0xf53ddf0 in /usr/bin/clickhouse
2021.03.11 09:06:57.695124 [ 8085 ] {} <Fatal> BaseDaemon: 6. DB::RemoteQueryExecutor::finish(std::__1::unique_ptr<DB::RemoteQueryExecutorReadContext, std::__1::default_delete<DB::RemoteQueryExecutorReadContext>
 >*) @ 0xe6b4657 in /usr/bin/clickhouse
2021.03.11 09:06:57.695420 [ 8085 ] {} <Fatal> BaseDaemon: 7. DB::PipelineExecutor::tryAddProcessorToStackIfUpdated(DB::ExecutingGraph::Edge&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::Execu
tingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, u
nsigned long) @ 0xf6527fc in /usr/bin/clickhouse
2021.03.11 09:06:57.695538 [ 8085 ] {} <Fatal> BaseDaemon: 8. DB::PipelineExecutor::prepareProcessor(unsigned long, unsigned long, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::N
ode*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::uni
que_lock<std::__1::mutex>) @ 0xf652fc9 in /usr/bin/clickhouse
2021.03.11 09:06:57.695673 [ 8085 ] {} <Fatal> BaseDaemon: 9. DB::PipelineExecutor::tryAddProcessorToStackIfUpdated(DB::ExecutingGraph::Edge&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::Execu
tingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, u
nsigned long) @ 0xf65289c in /usr/bin/clickhouse
2021.03.11 09:06:57.695779 [ 8085 ] {} <Fatal> BaseDaemon: 10. DB::PipelineExecutor::prepareProcessor(unsigned long, unsigned long, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::
Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::un
ique_lock<std::__1::mutex>) @ 0xf652fc9 in /usr/bin/clickhouse
2021.03.11 09:06:57.695882 [ 8085 ] {} <Fatal> BaseDaemon: 11. DB::PipelineExecutor::executeStepImpl(unsigned long, unsigned long, std::__1::atomic<bool>*) @ 0xf6564d0 in /usr/bin/clickhouse
2021.03.11 09:06:57.695965 [ 8085 ] {} <Fatal> BaseDaemon: 12. ? @ 0xf65aa06 in /usr/bin/clickhouse
2021.03.11 09:06:57.696360 [ 8085 ] {} <Fatal> BaseDaemon: 13. ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0x851158f in /usr/bin/clickhouse
2021.03.11 09:06:57.696426 [ 8085 ] {} <Fatal> BaseDaemon: 14. ? @ 0x8515023 in /usr/bin/clickhouse
2021.03.11 09:06:57.696583 [ 8085 ] {} <Fatal> BaseDaemon: 15. start_thread @ 0x7ea5 in /usr/lib64/libpthread-2.17.so
2021.03.11 09:06:57.696702 [ 8085 ] {} <Fatal> BaseDaemon: 16. clone @ 0xfe9fd in /usr/lib64/libc-2.17.so
2021.03.11 09:06:57.954654 [ 8085 ] {} <Fatal> BaseDaemon: Checksum of the binary: 3B52BD94C585A17D2C45E5ABAABC8124, integrity check passed.
```
But looks like this fatal was fixed in https://github.com/ClickHouse/ClickHouse/pull/20547

**Comment by filimonov** [comment]
Duplicate of #21167 ?

**Comment by SaltTan** [comment]
It's not a duplicate. There was a crash before, now it's just an exception.
I get it sometimes on 21.1.6 (which I installed to get rid of the crash).
Here is an example:
```
type                     | event_time_microseconds    | result_rows | exception_code | exception                                                                                                                                                         | is_initial_query
-------------------------+----------------------------+-------------+----------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------
QueryFinish              | 2021-03-14 15:32:06.914672 |      315259 |              0 |                                                                                                                                                                   |                0
QueryFinish              | 2021-03-14 15:32:06.929858 |      138217 |              0 |                                                                                                                                                                   |                0
ExceptionWhileProcessing | 2021-03-14 15:32:07.001291 |           0 |            210 | Code: 210, e.displayText() = DB::NetException: I/O error: Broken pipe, while writing to socket ([::ffff:10.246.194.3]:39622) (version 21.1.6.13 (official build)) |                0
QueryFinish              | 2021-03-14 15:32:07.027142 |      353308 |              0 |                                                                                                                                                                   |                0
QueryFinish              | 2021-03-14 15:32:07.051843 |      328086 |              0 |                                                                                                                                                                   |                0
ExceptionWhileProcessing | 2021-03-14 15:32:07.095584 |           0 |            100 | Code: 100, e.displayText() = DB::Exception: Unknown packet 48 from server he3:9000 (version 21.1.6.13 (official build))                                           |                1
```
Stack trace from the shard:
```
0. DB::WriteBufferFromPocoSocket::nextImpl() @ 0xeba2368 in /usr/bin/clickhouse
1. DB::TCPHandler::sendProgress() @ 0xf876976 in /usr/bin/clickhouse
2. DB::TCPHandler::processOrdinaryQueryWithProcessors() @ 0xf8740d5 in /usr/bin/clickhouse
3. DB::TCPHandler::runImpl() @ 0xf86d4cd in /usr/bin/clickhouse
4. DB::TCPHandler::run() @ 0xf87bbd7 in /usr/bin/clickhouse
5. Poco::Net::TCPServerConnection::start() @ 0x11dbfbcf in /usr/bin/clickhouse
6. Poco::Net::TCPServerDispatcher::run() @ 0x11dc15e1 in /usr/bin/clickhouse
7. Poco::PooledThread::run() @ 0x11eecf69 in /usr/bin/clickhouse
8. Poco::ThreadImpl::runnableEntry(void*) @ 0x11ee8efa in /usr/bin/clickhouse
9. start_thread @ 0x74a4 in /lib/x86_64-linux-gnu/libpthread-2.24.so
10. clone @ 0xe8d0f in /lib/x86_64-linux-gnu/libc-2.24.so
```
Stack trace from the initiator:
```
0. DB::Connection::receivePacket(std::__1::function<void (Poco::Net::Socket&)>) @ 0xf7cc369 in /usr/bin/clickhouse
1. DB::MultiplexedConnections::receivePacketUnlocked(std::__1::function<void (Poco::Net::Socket&)>) @ 0xf7db8f3 in /usr/bin/clickhouse
2. DB::MultiplexedConnections::drain() @ 0xf7dc027 in /usr/bin/clickhouse
3. DB::RemoteQueryExecutor::finish(std::__1::unique_ptr<DB::RemoteQueryExecutorReadContext, std::__1::default_delete<DB::RemoteQueryExecutorReadContext> >*) @ 0xe98a0a7 in /usr/bin/clickhouse
4. DB::PipelineExecutor::tryAddProcessorToStackIfUpdated(DB::ExecutingGraph::Edge&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, unsigned long) @ 0xf8f906c in /usr/bin/clickhouse
5. DB::PipelineExecutor::prepareProcessor(unsigned long, unsigned long, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::unique_lock<std::__1::mutex>) @ 0xf8f9839 in /usr/bin/clickhouse
6. DB::PipelineExecutor::tryAddProcessorToStackIfUpdated(DB::ExecutingGraph::Edge&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, unsigned long) @ 0xf8f910c in /usr/bin/clickhouse
7. DB::PipelineExecutor::prepareProcessor(unsigned long, unsigned long, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::unique_lock<std::__1::mutex>) @ 0xf8f9839 in /usr/bin/clickhouse
8. DB::PipelineExecutor::tryAddProcessorToStackIfUpdated(DB::ExecutingGraph::Edge&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, unsigned long) @ 0xf8f910c in /usr/bin/clickhouse
9. DB::PipelineExecutor::prepareProcessor(unsigned long, unsigned long, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::unique_lock<std::__1::mutex>) @ 0xf8f9839 in /usr/bin/clickhouse
10. DB::PipelineExecutor::tryAddProcessorToStackIfUpdated(DB::ExecutingGraph::Edge&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, unsigned long) @ 0xf8f910c in /usr/bin/clickhouse
11. DB::PipelineExecutor::prepareProcessor(unsigned long, unsigned long, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::unique_lock<std::__1::mutex>) @ 0xf8f9839 in /usr/bin/clickhouse
12. DB::PipelineExecutor::tryAddProcessorToStackIfUpdated(DB::ExecutingGraph::Edge&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, unsigned long) @ 0xf8f910c in /usr/bin/clickhouse
13. DB::PipelineExecutor::prepareProcessor(unsigned long, unsigned long, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::queue<DB::ExecutingGraph::Node*, std::__1::deque<DB::ExecutingGraph::Node*, std::__1::allocator<DB::ExecutingGraph::Node*> > >&, std::__1::unique_lock<std::__1::mutex>) @ 0xf8f97f9 in /usr/bin/clickhouse
14. DB::PipelineExecutor::executeStepImpl(unsigned long, unsigned long, std::__1::atomic<bool>*) @ 0xf8fcdf0 in /usr/bin/clickhouse
15. ? @ 0xf9018f5 in /usr/bin/clickhouse
16. ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0x863497d in /usr/bin/clickhouse
17. ? @ 0x8638533 in /usr/bin/clickhouse
18. start_thread @ 0x74a4 in /lib/x86_64-linux-gnu/libpthread-2.24.so
19. clone @ 0xe8d0f in /lib/x86_64-linux-gnu/libc-2.24.so
```

**Comment by filimonov** [comment]
@KochetovNicolai @azat any ideas?

**Comment by den-crane** [comment]
>SELECT  ORDER BY timestamp DESC LIMIT 100 FORMAT JSONCompact 

@SaltTan ? also has  HTTP  `SELECT ORDER BY ...  LIMIT ... FORMAT TabSeparatedWithNamesAndTypes`
Maybe it's related to LIMIT

**Comment by azat** [comment]
Should be fixed by #21643
